src/context.rs

```diff
@@ -192,15 +195,41 @@ pub struct Context<'a, 'b: 'a, 's> {
 
 ///A reader for a request body.
 pub struct BodyReader<'a, 'b: 'a> {
-    request: HttpReader<&'a mut BufReader<&'b mut NetworkStream>>
+    reader: HttpReader<&'a mut BufReader<&'b mut NetworkStream>>,
+    is_multipart: bool,
```

Comment:
Hmm... Does multipart even work without the boundary? If it doesn't, then you are right. Otherwise, I don't trust the request to include it.